src/integrationTest/java/com/mongodb/hibernate/BasicCRUDTests.java

```diff
@@ -170,6 +172,46 @@ void testSimpleDeletion() {
         }
     }
 
+    @Nested
+    class UpdateTests {
+
+        @ParameterizedTest(name = "merge: {0}")
+        @ValueSource(booleans = {true, false})
+        void testSimpleUpdate(boolean merge) {
+            var statementInspector = sessionFactoryScope.getStatementInspector(SQLStatementInspector.class);
+            statementInspector.clear();
+            sessionFactoryScope.inTransaction(session -> {
+                var book = new Book();
+                book.id = 1;
+                book.title = "War and Peace";
+                book.author = "Leo Tolstoy";
+                book.publishYear = 1867;
+                session.persist(book);
+                session.flush();
+
+                book.title = "Insurrection";
+                book.publishYear = 1899;
+                if (merge) {
+                    session.merge(book);
+                }
+            });
+
+            assertCollectionContainsOnly(
+                    BsonDocument.parse(
+                            """
+                            {"_id": 1, "author": "Leo Tolstoy", "publishYear": 1899, "title": "Insurrection"}\
+                            """));
+            assertThat(statementInspector.getSqlQueries())
+                    .containsExactly(
+                            """
+                            {"insert": "books", "documents": [{"author": {"$undefined": true}, "publishYear": {"$undefined": true}, "title": {"$undefined": true}, "_id": {"$undefined": true}}]}\
+                            """,
+                            """
+                            {"update": "books", "updates": [{"q": {"_id": {"$eq": {"$undefined": true}}}, "u": {"$set": {"author": {"$undefined": true}, "publishYear": {"$undefined": true}, "title": {"$undefined": true}}}, "multi": true}]}\
+                            """);
+        }
+    }
```

Comment:
to prove that the udpating did happen (not one sinlge insertion with latest state), a good feature of hibernate-testing was used here to verify. Note that by default table mutation static update SQL will include all fields to for perf reason

Reply:
I tried annotating `Book` with [`@DynamicUpdate`](https://docs.jboss.org/hibernate/orm/6.6/javadocs/org/hibernate/annotations/DynamicUpdate.html), and it just worked, barring the `statementInspector.getSqlQueries` assertion, of course. Should we add a test for that?

Reply:
good point. I think we should cover this. 

There is also a `@DynamicInsert`, which won't include fields with null values, so it will align with the null-free insertion case we discussed. I might add it later but not in this PR.